File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/mediator/Colleague.java
package jv.com.tencent.cain.design_pattern.mediator;

/**
 * @author cainjiang
 * @date 2020/8/13
 */
abstract class Colleague {
    protected Mediator mediator;

    public Colleague(Mediator mediator) {
        this.mediator = mediator;
    }

    abstract void notify(String msg);

    abstract void  send(String msg);

    abstract String  getName();
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/mediator/Colleague.kt
package kt.com.tencent.cain.desigin_pattern.mediator

/**
 * @author : jiangyu
 * @date   : 2020/8/13
 * @desc   : xxx
 */
abstract class Colleague(val mediator: Mediator) {
    abstract fun send(msg: String)
    abstract fun notify(msg: String)
    abstract fun getName(): String
}